spec/models/instrument_price_policy_calculations_spec.rb

```diff
@@ -16,18 +17,58 @@
     policy.calculate_cost_and_subsidy_from_order_detail fake_order_detail
   end
 
-  it "calculates cost based on the given duration and discount" do
-    policy.usage_rate = 5
-    expect(policy.calculate_cost(120, 0.15).round 2).to eq 1.5
+  describe "estimating cost and subsidy from an order detail" do
```

Comment:
Yeah, these are kind of higher-level than unit tests, especially now that I've introduced the additional class. I ended up writing more unit-level tests for that, and left most of these tests around largely as regression protection.